File: JavaAlgorithm/src/main/java/algo/tzashinorpu/ThirdRound/Chapter07/maxDepth_104.java
package algo.tzashinorpu.ThirdRound.Chapter07;

import algo.tzashinorpu.ThirdRound.TreeNode;

import java.util.ArrayDeque;
import java.util.Stack;

public class maxDepth_104 {
    public int maxDepth(TreeNode root) {
        int depth = 0;
        ArrayDeque<TreeNode> stack = new ArrayDeque<>();
        if (root != null)
            stack.push(root);
        while (!stack.isEmpty()) {
            int size = stack.size();
            for (int i = 0; i < size; i++) {
                TreeNode pop = stack.pollFirst();
                if (pop.left != null) {
                    stack.offerLast(pop.left);
                }
                if (pop.right != null) {
                    stack.offerLast(pop.right);
                }
            }
            depth++;
        }
        return depth;
    }
}

File: Clion/ZeroTrac/Tree/labuladong/maxDepth_104.cpp
//
// Created by TzashiNorpu on 3/13/2023.
//
#include "tree_r1.h"

using namespace Tree;

int Solution::maxDepth(TreeNode *root) {
  if (!root) return 0;
  int l = maxDepth(root->left);
  int r = maxDepth(root->right);
  return max(l, r) + 1;
}